magefile.go

```diff
@@ -281,39 +287,40 @@ func (b Build) ImageBuild(cmd string) error {
 		return b.buildPackageOperatorImage(imageCacheDir)
 
 	default:
-		mg.Deps(
+		deps := []interface{}{
 			mg.F(Build.cmd, cmd, "linux", "amd64"),
-		)
-		return b.buildGenericImage(cmd, imageCacheDir)
+			mg.F(populateCmdCache, imageCacheDir, cmd),
+		}
+		imageBuildInfo := newImageBuildInfo(cmd, imageCacheDir)
+		return dev.BuildImage(imageBuildInfo, deps)
 	}
 }
 
-// generic image build function, when the image just relies on
-// a static binary build from cmd/*
-func (Build) buildGenericImage(cmd, imageCacheDir string) error {
-	imageTag := imageURL(cmd)
-	for _, command := range [][]string{
-		// Copy files for build environment
-		{"cp", "-a",
-			"bin/linux_amd64/" + cmd,
-			imageCacheDir + "/" + cmd},
-		{"cp", "-a",
-			"config/docker/" + cmd + ".Dockerfile",
-			imageCacheDir + "/Dockerfile"},
-
-		// Build image!
-		{containerRuntime, "build", "-t", imageTag, imageCacheDir},
-		{containerRuntime, "image", "save",
-			"-o", imageCacheDir + ".tar", imageTag},
-	} {
+func populateCmdCache(imageCacheDir, cmd string) error {
```

Comment:
Why is this function called `populateCmdCache`? It seems to copy the artifacts from be previous build step?

Reply:
Hi, sorry for the late reply. I named all the functions which copy the necessary files to the directory, where the image is going to be built like this: `populate<type-of-image>Cache()`. I thought it made sense to group these functions together, since they essentially do the same thing, so I named them the same. Should I rename them somehow, or just this one?

Reply:
@ashishmax31 WDYT? :smiley:

Reply:
Aah yeah lets keep it as it as.